gui/src/2D/controls/structuredTextBlock.ts

```diff
@@ -0,0 +1,993 @@
+import { Observable } from "babylonjs/Misc/observable";
+import { Measure } from "../measure";
+import { ValueAndUnit } from "../valueAndUnit";
+import { Control } from "./control";
+import { TextWrapping } from "./textBlock";
+import { RegisterClass } from "babylonjs/Misc/typeStore";
+import { Nullable } from "babylonjs/types";
+import { serialize } from 'babylonjs/Misc/decorators';
+import { ICanvasRenderingContext , ICanvasGradient } from 'babylonjs/Engines/ICanvas';
+import { IStructuredTextPart } from './iStructuredTextPart';
+import { Engine } from 'babylonjs/Engines/engine';
+
+type StructuredText = Array<IStructuredTextPart>;
+
+type StructuredTextLine = {
+    parts: StructuredText;
+
+    // Computed line width
+    width?: number;
+};
+type StructuredTextLines = Array<StructuredTextLine>;
+
+// Mostly the same than IStructuredTextPart, but nothing is optional here,
+// this is the attributes about to be sent to the canvas context.
+interface TextPartAttributes {
```

Comment:
That should be a separated file as well ;)

Reply:
@deltakosh Is it really useful since this interface is only used internally?

Also maybe I should have used the `type` keyword instead of `interface` to better express this? There are still some subtleties between *type alias* and *interface* that I don't understand very well and [stackoverflow doesn't helped me much on that matter](https://stackoverflow.com/questions/37233735/interfaces-vs-types-in-typescript).